application/controllers/RedundancygroupController.php

```diff
@@ -174,19 +174,22 @@ public function childrenAction(): Generator
         );
         $viewModeSwitcher = $this->createViewModeSwitcher($paginationControl, $limitControl);
 
-        $searchBar = $this->createSearchBar(
-            $nodesQuery,
-            [
-                $limitControl->getLimitParam(),
-                $sortControl->getSortParam(),
-                $viewModeSwitcher->getViewModeParam(),
-                'id'
-            ]
-        );
-
-        $searchBar->getSuggestionUrl()->setParam('isChildrenTab');
-        $searchBar->getEditorUrl()
-            ->setParams((clone $searchBar->getEditorUrl()->getParams())->set('isChildrenTab', true));
+        $preserveParams = [
+            $limitControl->getLimitParam(),
+            $sortControl->getSortParam(),
+            $viewModeSwitcher->getViewModeParam(),
+            'id'
+        ];
+
+        $requestParams = Url::fromRequest()->onlyWith($preserveParams)->getParams();
+        $searchBar = $this->createSearchBar($nodesQuery, $preserveParams)
+            ->setEditorUrl(
+                Url::fromPath('icingadb/redundancygroup/children-search-editor')
+                    ->setParams($requestParams)
+            )->setSuggestionUrl(
+                Url::fromPath('icingadb/redundancygroup/children-complete')
+                    ->setParams($requestParams)
```

Comment:
You share one params object with two separate and very distinct url objects. I don't recommend this, clone it.